presto-rcfile/src/main/java/io/prestosql/rcfile/binary/TimestampEncoding.java

```diff
@@ -45,11 +43,10 @@
     private final DateTimeZone timeZone;
     private final PrestoTimestampEncoder<?> prestoTimestampEncoder;
 
-    public TimestampEncoding(Type type, DateTimeZone timeZone)
+    public TimestampEncoding(TimestampType type, DateTimeZone timeZone)
     {
         requireNonNull(type, "type is null");
-        verify(type instanceof TimestampType, "type is not a TimestampType");
-        this.type = (TimestampType) type;
+        this.type = requireNonNull(type, "type is null");
```

Comment:
`requireNonNull(type, "type is null")` is now checked twice

(i'd keep the one on the assigment line)